### ISSUE TITLE
Upload/analize fails without any reason

### ISSUE THREAD

**Issue by LNT00** [pandy06269/blue-twilight#74]
Is there a log file for failed uploads/analysis? I get random failures for no reason, on retry the files are uploaded and analyzed correctly.

**Comment by pandy06269** [comment]
I think this is related to the issue seen in your log files saying “mkdir: file exists”. Before the app calls mkdir it checks if the folder already exists and doesn’t try and create it if it does.

The fact mkdir is failing even though it shouldn’t be executed suggests a possible race condition with multiple uploads, explaining why it works if you try again later.

On my systems only errors are thrown however it looks like warnings are being thrown on your hosting causing the script to fail.

Can you please amend the `app\Helpers\FileHelper.php` file and replace “mkdir” with “@mkdir” which should suppress the warning and see if you still get failures?

**Comment by LNT00** [comment]
No, it's not related to that issue. I uploaded multiple files and some failed, without any warning. I will post here the next error i get.  When i retried only the failed photos, it worked correctly.
I couldn't find a specific case, I've uploaded file counts from 2 to 3600+ at one time, and nothing similar happened to get an idea why sometimes fail occurs.

**Comment by pandy06269** [comment]
Please try my suggested fix - this would cause random failures that seem to work fine the next time.

If there was an issue with a specific photo it would fail consistently.

I’ve uploaded 1,000 photos at a time myself and never had an issue because I have PHP warnings turned off.

**Comment by LNT00** [comment]
Sure, i already did that.

**Comment by andysh-uk** [comment]
I've added my suggested workaround to the 2.1.1 release. Please can you try this release once it's published (in the next couple of days) and see if you're still having the issue?
